Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Select.hs

```diff
@@ -120,6 +124,7 @@ select personId estimateId req@DEstimateSelectReq {..} = do
   estimate <- QEstimate.findById estimateId >>= fromMaybeM (EstimateDoesNotExist estimateId.getId)
   when (DEstimate.isCancelled estimate.status) $ throwError $ EstimateCancelled estimate.id.getId
   let searchRequestId = estimate.requestId
+  searchRetry <- SearchRetry.findById $ cast searchRequestId
```

Comment:
Using `cast` function is not very safe because we can confuse `Id`s from different entities. Can we put  instead:
`id SearchRequestTId` -- on tabular level

`id :: Id SearchRequest` -- on domain level
so that we don't need to use `cast`.

Similar logic we already have in our project, for example `DriverInformation` entity use `Id Person`

Reply:
done